### ISSUE TITLE
tasks.startTask has incorrect type for submitHandler

### ISSUE THREAD

**Issue by navzam** [OfficeDev/microsoft-teams-library-js#444]
`microsoftTeams.tasks.startTask()` accepts a `submitHandler` that is called when the task module closes. The type of `submitHandler` is
```javascript
(err: string, result: string) => void)
```
But based on the [task module docs](https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/task-modules/task-modules-tabs#example-submitting-the-result-of-a-task-module) and the type of [`submitTask()`](https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/microsoftteams.tasks?view=msteams-client-js-latest#submittask-string---object--string---string---), `result` can be a `string` or an `object`. I thought maybe the object always gets serialized before being passed to `submitHandler`, but that doesn't seem to be the case. I can call `submitTask()` with an object, and `result` comes through as an object.

So shouldn't `submitHandler`'s `result` param be of type `string | object` ?

**Comment by Nikitha-MSFT** [comment]
You need to pass the Object to submithandler's result. The object contains the string

**Comment by navzam** [comment]
The functionality works fine. I can pass an object in `submitTask()` and the object comes through in the `submitHandler`. I'm more talking about the TypeScript typings, which indicates that the handler's `result` can only be a `string`, when really it could be an `object`:

![submit_handler](https://user-images.githubusercontent.com/126523/97934859-bdcc1580-1d2b-11eb-8295-da9ab943fc37.PNG)

The `submitTask()` typings seem correct:

![submit_task](https://user-images.githubusercontent.com/126523/97934877-cf152200-1d2b-11eb-909a-168ea6744e98.PNG)

**Comment by Nikitha-MSFT** [comment]
Object is a set of strings when you pass an object typescript will consider that as a string, hence the result is shown as string. This is by design.

**Comment by navzam** [comment]
> when you pass an object typescript will consider that as a string, hence the result is shown as string

This isn't the behavior I'm seeing. In `submitTask` I'm passing an object (as allowed by the SDK's typings and shown in the [task module docs](https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/task-modules/task-modules-tabs#example-submitting-the-result-of-a-task-module)):

```ts
microsoftTeams.tasks.submitTask({ foo: 'bar' });
```

and in the `submitHandler`, if I try to use it as a string:

```ts
microsoftTeams.tasks.startTask(taskInfo, (err, result) => {
    console.log('Got result of assign task module');
    console.log(result);
    console.log(result.toUpperCase()); // this is allowed by TS because the SDK's type of result is string
});
```

The console shows an error about `toUpperCase` (because `result` is not a string):

![console_log_output](https://user-images.githubusercontent.com/126523/99406514-cfbcc500-28a2-11eb-8c3f-33dfc53b1570.PNG)

The [task module docs](https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/task-modules/task-modules-tabs#example-invoking-a-task-module) also show `result` being used as an object.

**Comment by Nikitha-MSFT** [comment]
Raised a bug- we don't have ETA to share when it will be fixed.

**Comment by MarcinJuraszek** [comment]
Not sure about fixing it in the existing 1.* version, but we'll definitely want to fix that for 2.0-preview release. I sent a PR to address that there. Thank you for reporting the bug, @navzam!

**Comment by ChetanSharma-msft** [comment]
Hello @navzam - Hope you are doing well!!

Looks like, this issue has fixed and already merged.

We are closing this issue for now, please feel free to reopen it, if required.
